src/test/java/org/folio/rest/api/PreviouslyHeldDataUpgradeTest.java

```diff
@@ -0,0 +1,66 @@
+package org.folio.rest.api;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.isNull;
+
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.folio.rest.persist.PostgresClient;
+import org.folio.util.ResourceUtil;
+import io.vertx.core.json.JsonObject;
+
+import io.vertx.core.Vertx;
+import junitparams.JUnitParamsRunner;
+
+@RunWith(JUnitParamsRunner.class)
+public class PreviouslyHeldDataUpgradeTest extends TestBaseWithInventoryUtil {
+    private static Vertx vertx = Vertx.vertx();
+    private static UUID instanceId = UUID.randomUUID();
+    private static final String SET_DEFAULT_PREVIOUSLY_HELD = ResourceUtil
+        .asString("templates/db_scripts/setPreviouslyHeldDefault.sql")
+        .replace("${myuniversity}_${mymodule}", "test_tenant_mod_inventory_storage");
+
+    @Test
+    public void canMigrateToDefaultPreviouslyHeldValue() {
+        instancesClient.create(instance(instanceId));
+
+        String query = "UPDATE test_tenant_mod_inventory_storage.instance " +
+            "SET jsonb = JSONB_SET(instance.jsonb, '{previouslyHeld}', TO_JSONB(null)) " +
+            "WHERE jsonb->>'previouslyHeld' IS NOT NULL;";
```

Comment:
Could this approach work instead - https://til.hashrocket.com/posts/78333d571f-remove-keyvalue-from-jsonb-data-structure ?

Reply:
I spent most of the morning trying this-I either get syntax errors or it seems to execute, but do nothing.  Closest I've gotten to success is `UPDATE test_tenant_mod_inventory_storage.instance SET jsonb = jsonb - 'previouslyHeld';`
This executes without an sql error, but when I check the previouslyHeld key afterwards, it's still present and still set to false.

Reply:
Thanks for the update. I don't think it is worth trying to do test this more thoroughly given these challenges.

Have / Could you test this manually as a one off?

Reply:
What if I merge this without the test and add it later, once I've figured out how it's supposed to work?  I can ask Bohdan or some of the other back-end devs for help.  This is going to come up again for https://issues.folio.org/browse/MODINVSTOR-669, as that issue will also need a migration script, so I'd like to know what I'm doing wrong.  

I have not tested it manually, and I might need some guidance on how to do so-is there a script I can invoke from the command-line that will set up the database?  If I can do that, I could presumably run the sql script from the database command line.  I'm assuming I'll need to install postgresql locally?

Reply:
> What if I merge this without the test and add it later, once I've figured out how it's supposed to work?

Sure, I'm ok with that. @oleksandr-yatsenko1 has been doing some work on testing upgrades for his pull request, there might be some overlap.

> I might need some guidance on how to do so-is there a script I can invoke from the command-line that will set up the database? 

There are some scripts in the repository. I might be the only person who uses them.

[This](https://wiki.folio.org/display/FOLIJET/How+to+test+data+migration+scripts+locally) is the best documentation we have at the moment.

> If I can do that, I could presumably run the sql script from the database command line. I'm assuming I'll need to install postgresql locally?

Yes, or use a docker image.

I've been working on a Docker compose file for setting up FOLIO infrastructure and running a module.

Reply:
I asked @bohdan-suprun about this, he says what's happening is when I call the api, the JSON data is deserialized into a java object, and because I've set a default value for this attribute in the RAML, it's automatically populated at that point if it's null.  So I am actually erasing the value, I'm just automatically re-setting it when I call the API to retrieve it.  He recommends pulling the data directly from the database to make sure it's been set properly rather than calling the API, so that's what I'm going to try and do.

Reply:
> what's happening is when I call the api, the JSON data is deserialized into a java object, and because I've set a default value for this attribute in the RAML, it's automatically populated at that point if it's null

That seems totally reasonable. It's one of the reasons I'm not keen on this code generation approach.

I think that means we cannot easily test the upgrade process.

Reply:
> He recommends pulling the data directly from the database to make sure it's been set properly rather than calling the API, so that's what I'm going to try and do.

That is worth trying

Reply:
Okay, I think I finally have this working.